Controllers/PurchasesController.cs

```diff
@@ -97,30 +112,52 @@ public IActionResult Create()
             var purchaseVM = new PurchaseFormViewModel
             {
                 Branches = _context.Branches.ToList(),
-                Customers = _context.Customers.ToList()
+                Customers = _context.Customers.ToList(),
+                Products = _context.Products.ToList()
             };
 
             return View(purchaseVM);
         }
 
         // POST: Purchases/Create
         [HttpPost]
-        [ValidateAntiForgeryToken]
-        public async Task<IActionResult> Create(PurchaseFormViewModel purchaseVM)
+        // [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([FromBody] PurchaseFormViewModel purchaseVM)
         {
             if (ModelState.IsValid)
             {
                 var branch = await _context.Branches
                     .SingleOrDefaultAsync(m => m.Id == purchaseVM.BranchId);
+
                 purchaseVM.EntityId = branch.EntityId;
+                purchaseVM.Identification = DateTime.Now.ToString("yyMMddHHmmss0000ffff");
```

Comment:
For this purpose better seems to be 
`purchaseVM.Identification = Guid.NewGuid()`
as now there is risk(minimal) to have multiplicates purchases (paraller users). This risk can be removed with Guid generator. 

https://docs.microsoft.com/pl-pl/dotnet/api/system.guid.newguid?view=netcore-3.1

Reply:
you can also do some combination `$"{DateTime.Now.ToString("yyMMddHHmmss0000ffff")}-{Guid.NewGuid()}"`

Reply:
This field will be not used as ID, but it is good to maintain its integrity. 
Is there a way to assure the uniqueness using only numbers?
So far I changed to **DateTime.Now.Ticks + randonNum.Next(1, 9999999)**
Thank you, good to know this GUID, I will keep in mind to use in the future.